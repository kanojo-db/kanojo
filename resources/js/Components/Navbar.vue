<script setup lang="ts">
import { Link, router } from '@inertiajs/vue3';
import { computed, ref, watch } from 'vue';
import { useDisplay } from 'vuetify';

import MdiMagnify from '~icons/mdi/magnify';
import MdiMenu from '~icons/mdi/menu';
import MdiPlus from '~icons/mdi/plus';

import LanguageSwitcher from '@/Components/LanguageSwitcher.vue';
import useRouteStore from '@/stores/route';
import { getMenuEntries } from '@/utils/misc';

import UserMenu from './UserMenu.vue';

const props = defineProps({
    showDrawer: {
        type: Boolean,
    },
});

const emit = defineEmits<{
    (event: 'update:showSearch', value: boolean): void;
    (event: 'update:showDrawer', value: boolean): void;
}>();

const menuEntries = getMenuEntries();

const routeStore = useRouteStore();

const searchParam = computed(() => routeStore.params?.q);

const search = ref('');

const submit = () => {
    router.get(
        route('search', {
            q: search.value,
        }),
    );
};

watch(
    () => searchParam,
    () => {
        search.value = searchParam.value ?? '';
    },
    { immediate: true },
);

const manuallyShowSearch = ref(false);
const showSearch = computed(() => !!searchParam.value);

const { mdAndUp } = useDisplay();
</script>

<template>
    <v-app-bar
        flat
        :extension-height="
            manuallyShowSearch || showSearch || $page.component === 'Search'
                ? 56
                : 0
        "
        class="bg-primary text-stone-50"
    >
        <v-app-bar-nav-icon
            class="block md:hidden"
            @click="() => emit('update:showDrawer', !props.showDrawer)"
        >
            <mdi-menu />
        </v-app-bar-nav-icon>

        <v-app-bar-title class="mr-2 max-w-[130px]">
            <Link :href="route('welcome')">
                <img
                    alt="Kanojo"
                    src="/images/logo-dark.svg"
                    class="h-[36px] w-[130px] object-contain"
                />
            </Link>
        </v-app-bar-title>

        <v-toolbar-items class="hidden w-fit gap-4 md:block">
            <v-menu
                v-for="(menuEntry, index) in menuEntries"
                :key="`nav-menu-${index}`"
                open-on-hover
            >
                <template #activator="{ props: menuProps }">
                    <v-btn
                        v-bind="menuProps"
                        variant="plain"
                        :text="menuEntry.title"
                    />
                </template>

                <v-list>
                    <v-list-item
                        v-for="(subMenuItem, subIndex) in menuEntry.items"
                        :key="`nav-menu-${index}-sub-${subIndex}`"
                        :to="
                            route(
                                subMenuItem.routeName,
                                subMenuItem.routeParams,
                            )
                        "
                    >
                        {{ subMenuItem.title }}
                    </v-list-item>
                </v-list>
            </v-menu>
        </v-toolbar-items>

        <template #append>
            <div class="flex flex-row items-center gap-4">
                <v-menu
                    v-if="mdAndUp"
                    open-on-hover
                    offset="8px"
                >
                    <template #activator="{ props: menuProps }">
                        <v-btn
                            v-bind="menuProps"
                            rounded
                            variant="plain"
                            :icon="MdiPlus"
                            aria-label="Add"
                        />
                    </template>

                    <v-list>
                        <v-list-item :to="route('movies.create')">
                            {{ $t('general.addMovie') }}
                        </v-list-item>

                        <v-list-item :to="route('models.create')">
                            {{ $t('general.addModel') }}
                        </v-list-item>

                        <v-list-item :to="route('studios.create')">
                            {{ $t('general.addStudio') }}
                        </v-list-item>

                        <v-list-item :to="route('series.create')">
                            {{ $t('general.addSeries') }}
                        </v-list-item>
                    </v-list>
                </v-menu>

                <language-switcher />

                <user-menu />

                <v-btn
                    variant="plain"
                    :icon="MdiMagnify"
                    :aria-label="`Open search`"
                    @click="() => (manuallyShowSearch = !manuallyShowSearch)"
                />
            </div>
        </template>

        <template #extension>
            <v-text-field
                v-if="
                    manuallyShowSearch ||
                    showSearch ||
                    $page.component === 'Search'
                "
                v-model="search"
                :rounded="0"
                variant="solo"
                hide-details
                :placeholder="$t('general.search_placeholder')"
                @keyup.enter="submit"
            >
                <template #append-inner>
                    <v-btn
                        color="primary"
                        text="Search"
                        @click="submit"
                    />
                </template>
            </v-text-field>
        </template>
    </v-app-bar>
</template>
