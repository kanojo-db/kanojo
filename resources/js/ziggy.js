const Ziggy = {"url":"http:\/\/192.168.0.44","port":null,"defaults":{},"routes":{"scribe":{"uri":"docs","methods":["GET","HEAD"]},"scribe.postman":{"uri":"docs.postman","methods":["GET","HEAD"]},"scribe.openapi":{"uri":"docs.openapi","methods":["GET","HEAD"]},"dusk.login":{"uri":"_dusk\/login\/{userId}\/{guard?}","methods":["GET","HEAD"]},"dusk.logout":{"uri":"_dusk\/logout\/{guard?}","methods":["GET","HEAD"]},"dusk.user":{"uri":"_dusk\/user\/{guard?}","methods":["GET","HEAD"]},"login":{"uri":"login","methods":["GET","HEAD"]},"logout":{"uri":"logout","methods":["POST"]},"password.request":{"uri":"forgot-password","methods":["GET","HEAD"]},"password.reset":{"uri":"reset-password\/{token}","methods":["GET","HEAD"]},"password.email":{"uri":"forgot-password","methods":["POST"]},"password.update":{"uri":"reset-password","methods":["POST"]},"register":{"uri":"register","methods":["GET","HEAD"]},"verification.notice":{"uri":"email\/verify","methods":["GET","HEAD"]},"verification.verify":{"uri":"email\/verify\/{id}\/{hash}","methods":["GET","HEAD"]},"verification.send":{"uri":"email\/verification-notification","methods":["POST"]},"user-profile-information.update":{"uri":"user\/profile-information","methods":["PUT"]},"user-password.update":{"uri":"user\/password","methods":["PUT"]},"password.confirmation":{"uri":"user\/confirmed-password-status","methods":["GET","HEAD"]},"password.confirm":{"uri":"user\/confirm-password","methods":["POST"]},"horizon.stats.index":{"uri":"horizon\/api\/stats","methods":["GET","HEAD"]},"horizon.workload.index":{"uri":"horizon\/api\/workload","methods":["GET","HEAD"]},"horizon.masters.index":{"uri":"horizon\/api\/masters","methods":["GET","HEAD"]},"horizon.monitoring.index":{"uri":"horizon\/api\/monitoring","methods":["GET","HEAD"]},"horizon.monitoring.store":{"uri":"horizon\/api\/monitoring","methods":["POST"]},"horizon.monitoring-tag.paginate":{"uri":"horizon\/api\/monitoring\/{tag}","methods":["GET","HEAD"]},"horizon.monitoring-tag.destroy":{"uri":"horizon\/api\/monitoring\/{tag}","methods":["DELETE"],"wheres":{"tag":".*"}},"horizon.jobs-metrics.index":{"uri":"horizon\/api\/metrics\/jobs","methods":["GET","HEAD"]},"horizon.jobs-metrics.show":{"uri":"horizon\/api\/metrics\/jobs\/{id}","methods":["GET","HEAD"]},"horizon.queues-metrics.index":{"uri":"horizon\/api\/metrics\/queues","methods":["GET","HEAD"]},"horizon.queues-metrics.show":{"uri":"horizon\/api\/metrics\/queues\/{id}","methods":["GET","HEAD"]},"horizon.jobs-batches.index":{"uri":"horizon\/api\/batches","methods":["GET","HEAD"]},"horizon.jobs-batches.show":{"uri":"horizon\/api\/batches\/{id}","methods":["GET","HEAD"]},"horizon.jobs-batches.retry":{"uri":"horizon\/api\/batches\/retry\/{id}","methods":["POST"]},"horizon.pending-jobs.index":{"uri":"horizon\/api\/jobs\/pending","methods":["GET","HEAD"]},"horizon.completed-jobs.index":{"uri":"horizon\/api\/jobs\/completed","methods":["GET","HEAD"]},"horizon.silenced-jobs.index":{"uri":"horizon\/api\/jobs\/silenced","methods":["GET","HEAD"]},"horizon.failed-jobs.index":{"uri":"horizon\/api\/jobs\/failed","methods":["GET","HEAD"]},"horizon.failed-jobs.show":{"uri":"horizon\/api\/jobs\/failed\/{id}","methods":["GET","HEAD"]},"horizon.retry-jobs.show":{"uri":"horizon\/api\/jobs\/retry\/{id}","methods":["POST"]},"horizon.jobs.show":{"uri":"horizon\/api\/jobs\/{id}","methods":["GET","HEAD"]},"horizon.index":{"uri":"horizon\/{view?}","methods":["GET","HEAD"],"wheres":{"view":"(.*)"}},"sanctum.csrf-cookie":{"uri":"sanctum\/csrf-cookie","methods":["GET","HEAD"]},"ignition.healthCheck":{"uri":"_ignition\/health-check","methods":["GET","HEAD"]},"ignition.executeSolution":{"uri":"_ignition\/execute-solution","methods":["POST"]},"ignition.updateConfig":{"uri":"_ignition\/update-config","methods":["POST"]},"debugbar.openhandler":{"uri":"_debugbar\/open","methods":["GET","HEAD"]},"debugbar.clockwork":{"uri":"_debugbar\/clockwork\/{id}","methods":["GET","HEAD"]},"debugbar.assets.css":{"uri":"_debugbar\/assets\/stylesheets","methods":["GET","HEAD"]},"debugbar.assets.js":{"uri":"_debugbar\/assets\/javascript","methods":["GET","HEAD"]},"debugbar.cache.delete":{"uri":"_debugbar\/cache\/{key}\/{tags?}","methods":["DELETE"]},"welcome":{"uri":"\/","methods":["GET","HEAD"]},"privacy":{"uri":"privacy-policy","methods":["GET","HEAD"]},"login.provider":{"uri":"login\/{provider}","methods":["GET","HEAD"]},"login.provider.callback":{"uri":"login\/{provider}\/callback","methods":["GET","HEAD"]},"about.index":{"uri":"about","methods":["GET","HEAD"]},"settings.account":{"uri":"settings\/account","methods":["GET","HEAD"]},"settings.account.update":{"uri":"settings\/account","methods":["POST"]},"settings.sessions":{"uri":"settings\/sessions","methods":["GET","HEAD"]},"settings.tokens":{"uri":"settings\/tokens","methods":["GET","HEAD"]},"settings.tokens.create":{"uri":"settings\/tokens","methods":["POST"]},"settings.tokens.destroy":{"uri":"settings\/tokens\/{token}","methods":["DELETE"]},"search":{"uri":"search","methods":["GET","HEAD"]},"movies.index":{"uri":"movies","methods":["GET","HEAD"]},"movies.create":{"uri":"movies\/create","methods":["GET","HEAD"]},"movies.store":{"uri":"movies","methods":["POST"]},"movies.show":{"uri":"movies\/{movie}","methods":["GET","HEAD"]},"movies.edit":{"uri":"movies\/{movie}\/edit","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.update":{"uri":"movies\/{movie}","methods":["PUT","PATCH"],"bindings":{"movie":"slug"}},"movies.destroy":{"uri":"movies\/{movie}","methods":["DELETE"],"bindings":{"movie":"slug"}},"movies.media.index":{"uri":"movies\/{movie}\/media","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.media.store":{"uri":"movies\/{movie}\/media","methods":["POST"],"bindings":{"movie":"slug"}},"media.destroy":{"uri":"media\/{medium}","methods":["DELETE"]},"movies.history.index":{"uri":"movies\/{movie}\/history","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.like.store":{"uri":"movies\/{movie}\/like","methods":["POST"],"bindings":{"movie":"slug"}},"movies.dislike.store":{"uri":"movies\/{movie}\/dislike","methods":["POST"],"bindings":{"movie":"slug"}},"movies.favorite.store":{"uri":"movies\/{movie}\/favorite","methods":["POST"],"bindings":{"movie":"slug"}},"favorite.destroy":{"uri":"favorite\/{favorite}","methods":["DELETE"]},"movies.wishlist.store":{"uri":"movies\/{movie}\/wishlist","methods":["POST"],"bindings":{"movie":"slug"}},"wishlist.destroy":{"uri":"wishlist\/{wishlist}","methods":["DELETE"]},"movies.collection.store":{"uri":"movies\/{movie}\/collection","methods":["POST"],"bindings":{"movie":"slug"}},"collection.destroy":{"uri":"collection\/{collection}","methods":["DELETE"]},"movies.reports.index":{"uri":"movies\/{movie}\/reports","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.reports.create":{"uri":"movies\/{movie}\/reports\/create","methods":["GET","HEAD"]},"movies.reports.store":{"uri":"movies\/{movie}\/reports","methods":["POST"],"bindings":{"movie":"slug"}},"reports.show":{"uri":"reports\/{report}","methods":["GET","HEAD"]},"reports.edit":{"uri":"reports\/{report}\/edit","methods":["GET","HEAD"]},"reports.update":{"uri":"reports\/{report}","methods":["PUT","PATCH"]},"reports.destroy":{"uri":"reports\/{report}","methods":["DELETE"]},"models.index":{"uri":"models","methods":["GET","HEAD"]},"models.create":{"uri":"models\/create","methods":["GET","HEAD"]},"models.store":{"uri":"models","methods":["POST"]},"models.show":{"uri":"models\/{model}","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.edit":{"uri":"models\/{model}\/edit","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.update":{"uri":"models\/{model}","methods":["PUT","PATCH"],"bindings":{"model":"id"}},"models.destroy":{"uri":"models\/{model}","methods":["DELETE"]},"models.media.index":{"uri":"models\/{model}\/media","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.media.store":{"uri":"models\/{model}\/media","methods":["POST"],"bindings":{"model":"id"}},"models.history.index":{"uri":"models\/{model}\/history","methods":["GET","HEAD"],"bindings":{"model":"id"}},"studios.index":{"uri":"studios","methods":["GET","HEAD"]},"studios.create":{"uri":"studios\/create","methods":["GET","HEAD"]},"studios.store":{"uri":"studios","methods":["POST"]},"studios.show":{"uri":"studios\/{studio}","methods":["GET","HEAD"],"bindings":{"studio":"id"}},"studios.edit":{"uri":"studios\/{studio}\/edit","methods":["GET","HEAD"],"bindings":{"studio":"id"}},"studios.update":{"uri":"studios\/{studio}","methods":["PUT","PATCH"],"bindings":{"studio":"id"}},"studios.destroy":{"uri":"studios\/{studio}","methods":["DELETE"],"bindings":{"studio":"id"}},"profile.show":{"uri":"user\/{user}","methods":["GET","HEAD"],"bindings":{"user":"id"}},"bible.general":{"uri":"bible\/general","methods":["GET","HEAD"]},"user.locale.store":{"uri":"user\/locale","methods":["POST"]}}};

if (typeof window !== 'undefined' && typeof window.Ziggy !== 'undefined') {
    Object.assign(Ziggy.routes, window.Ziggy.routes);
}

export { Ziggy };
