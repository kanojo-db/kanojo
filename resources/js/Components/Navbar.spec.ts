/**
 * @vitest-environment jsdom
 */
import { createTestingPinia } from '@pinia/testing';
import { mount, renderToString } from '@vue/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import { describe, expect, it } from 'vitest';

import Navbar from '@/Components/Navbar.vue';
import getI18nPlugin from '@/plugins/i18n';
import getVuetifyPlugin from '@/plugins/vuetify';
import { ziggyProps } from '@/test/ziggy.mock';

expect.extend(toHaveNoViolations);

const i18n = getI18nPlugin('en-US');
const vuetify = getVuetifyPlugin(i18n);

vi.mock('@inertiajs/vue3', async () => {
    const mod =
        await vi.importActual<typeof import('@inertiajs/vue3')>(
            '@inertiajs/vue3',
        );

    return {
        ...mod,
        usePage: () => ({
            props: {
                value: {
                    ...ziggyProps,
                },
            },
        }),
    };
});

describe('Navbar', () => {
    it('renders properly in SSR mode', async () => {
        const contents = await renderToString(
            {
                template: '<v-app><v-layout><navbar /></v-layout></v-app>',
                components: {
                    Navbar,
                },
            },
            {
                global: {
                    plugins: [i18n, vuetify, createTestingPinia()],
                    mocks: {
                        page: {
                            props: {
                                ...ziggyProps,
                            },
                        },
                        $page: {
                            component: 'Welcome',
                        },
                    },
                },
            },
        );

        // Expect renderToString to return a string
        expect(typeof contents).toBe('string');
        // Expect renderToString to not throw any errors
        expect(contents).toMatchSnapshot();
    });

    it('has no a11y violations', async () => {
        const wrapper = mount(
            {
                template: '<v-app><v-layout><navbar /></v-layout></v-app>',
                components: {
                    Navbar,
                },
            },
            {
                global: {
                    mocks: {
                        $route: vi.fn(),
                        page: {
                            props: {
                                ...ziggyProps,
                            },
                        },
                        $page: {
                            component: 'Welcome',
                        },
                    },
                    plugins: [i18n, vuetify, createTestingPinia()],
                },
            },
        );

        const results = await axe(wrapper.html());

        expect(results).toHaveNoViolations();
    });
});
