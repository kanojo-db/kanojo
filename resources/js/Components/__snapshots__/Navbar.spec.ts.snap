// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders properly in SSR mode 1`] = `"<div class=\\"v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr\\" style=\\"z-index:1000;\\"><div class=\\"v-application__wrap\\"><!--[--><div class=\\"v-layout\\" style=\\"z-index:900;position:relative;overflow:hidden;\\"><!--[--><header class=\\"v-toolbar v-toolbar--flat v-toolbar--density-default v-theme--light v-locale--is-ltr v-app-bar bg-primary text-stone-50\\" style=\\"top:0;z-index:904;transform:translateY(0%);position:absolute;transition:none !important;\\"><!----><!--[--><div class=\\"v-toolbar__content\\" style=\\"height:64px;\\"><!----><!----><!--[--><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text block md:hidden v-app-bar-nav-icon\\" style=\\"\\" aria-label=\\"Menu\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><i class=\\"v-icon notranslate v-theme--light v-icon--size-default\\" style=\\"\\" aria-hidden=\\"true\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\"><path fill=\\"currentColor\\" d=\\"M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z\\"></path></svg></i></span><!----><!----></button><div class=\\"v-toolbar-title mr-2 max-w-[130px] v-app-bar-title\\" style=\\"\\"><div class=\\"v-toolbar-title__placeholder\\"><!----><!--[--><a href=\\"http://192.168.0.44/\\"><img alt=\\"Kanojo\\" src=\\"/images/logo-dark.svg\\" class=\\"h-[36px] w-[130px] object-contain\\"></a><!--]--></div></div><div class=\\"v-toolbar-items hidden w-fit gap-4 md:block\\" style=\\"\\"><!--[--><!--[--><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain\\" style=\\"height:inherit;\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-2\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[-->Movies<!--]--></span><!----><!----></button><!--]--><!----><!--]--><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain\\" style=\\"height:inherit;\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-4\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[-->Models<!--]--></span><!----><!----></button><!--]--><!----><!--]--><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain\\" style=\\"height:inherit;\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-6\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[-->Studios<!--]--></span><!----><!----></button><!--]--><!----><!--]--><!--]--><!--]--></div><!--]--><div class=\\"v-toolbar__append\\"><!--[--><div class=\\"flex flex-row items-center gap-4\\"><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--rounded v-btn--size-default v-btn--variant-plain\\" style=\\"\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-8\\" aria-label=\\"Add\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><i class=\\"v-icon notranslate v-theme--light v-icon--size-default\\" style=\\"\\" aria-hidden=\\"true\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\"><path fill=\\"currentColor\\" d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2Z\\"></path></svg></i></span><!----><!----></button><!--]--><!----><!--]--><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal rounded-none font-black\\" style=\\"\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-10\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[-->EN<!--]--></span><!----><!----></button><!--]--><!----><!--]--><!--[--><a class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain hidden lg:flex\\" style=\\"\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[--><!--[-->Login<!--]--><!--]--></span><!----><!----></a><a class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain hidden lg:flex\\" style=\\"\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><!--[--><!--[-->Register<!--]--><!--]--></span><!----><!----></a><!--[--><!--[--><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text lg:hidden\\" style=\\"\\" aria-label=\\"Open user menu\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-14\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><i class=\\"v-icon notranslate v-theme--light v-icon--size-default\\" style=\\"\\" aria-hidden=\\"true\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\"><path fill=\\"currentColor\\" d=\\"M12 4a4 4 0 0 1 4 4a4 4 0 0 1-4 4a4 4 0 0 1-4-4a4 4 0 0 1 4-4m0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4Z\\"></path></svg></i></span><!----><!----></button><!--]--><!----><!--]--><!--]--><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain\\" style=\\"\\" aria-label=\\"Open search\\"><!--[--><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><!--]--><!----><span class=\\"v-btn__content\\" data-no-activator><i class=\\"v-icon notranslate v-theme--light v-icon--size-default\\" style=\\"\\" aria-hidden=\\"true\\"><svg viewBox=\\"0 0 24 24\\" width=\\"1.2em\\" height=\\"1.2em\\"><path fill=\\"currentColor\\" d=\\"M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5l-1.5 1.5l-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16A6.5 6.5 0 0 1 3 9.5A6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14S14 12 14 9.5S12 5 9.5 5Z\\"></path></svg></i></span><!----><!----></button></div><!--]--></div></div><!--]--><!--[--><transition-stub name=\\"expand-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\"><div class=\\"v-toolbar__extension\\" style=\\"height:0px;\\"><!--[--><!--v-if--><!--]--></div></transition-stub><!--]--></header><!--]--></div><!--]--></div></div>"`;
